sympy/geometry/shape.py

```diff
@@ -0,0 +1,318 @@
+""" 2D geometrical entity constructed by boolean operations
+    of various geometric elements.
+
+"""
+
+from __future__ import division, print_function
+from sympy.core import S, pi, sympify
+from sympy.core.compatibility import ordered
+from sympy.core.symbol import _symbol
+from sympy.core.numbers import Rational, oo
+from sympy import symbols, simplify, solve
+from sympy.geometry.entity import GeometryEntity, GeometrySet
+from sympy.geometry.point import Point, Point2D
+from sympy.geometry.line import Line, Line2D, Ray2D, Segment2D, LinearEntity3D
+from sympy.geometry.ellipse import Ellipse, Circle
+from sympy.geometry.parabola import Parabola
+from sympy.functions.elementary.trigonometric import asin
+
+from sympy import Abs, sqrt
+
+class Shape(GeometrySet):
+    """
+    Parameters
+    ==========
+
+    conic : Ellipse, Circle, Parabola
+    line : Line
+
+    Attributes
+    ==========
+
+    area
+    centroid
+    second moment of area
+    product moment of area
+
+    Raises
+    ======
+
+    TypeError
+        Wrong type of argument were put
+        When The generated shape is not a closed figure
+    NotImplementedError
+        When `line` is neither horizontal nor vertical.
+
+    Notes
+    =====
+
+    If the conic figure is Circle or Ellipse and if the line is vertical,
+    then the shape will be segment that belongs to right side of the line.
+    And if the line is horizontal, then the shape will be above the line.
```

Comment:
I'm not really clear what this class is supposed to represent, but this seems confusing to me. Is there a better way to construct such shapes?

Reply:
![screenshot from 2018-03-23 17-38-31](https://user-images.githubusercontent.com/23614977/37828395-1387c3d6-2ec1-11e8-9686-6e84002a1677.png)
![screenshot from 2018-03-23 17-33-52](https://user-images.githubusercontent.com/23614977/37828396-13becc32-2ec1-11e8-8662-38b7690b780e.png)

It will give the property of the shaded region and this is form by the intersection of the ellipse with vertical and horizontal lines.

Reply:
we  can form similar type of close shape using straight line and  (parabola, ellipse, circle)

Reply:
> not really clear what this class is supposed to represent, but this seems confusing to me. Is there a better way to construct such shapes?

These might better be termed "segments" of conics (and I would prefer that to "shape" which is a much more general thing). They could perhaps be constructed by giving a line, width, and height and then another parameter that would determine (when width != height) whether the segment is for a parabola or an ellipse. But using the conic itself doesn't seem too bad.